ocaml/xapi/xapi_pool_update.ml

```diff
@@ -146,15 +166,12 @@ let detach_helper ~__context ~uuid ~vdi =
           ("detach_helper: unmounting " ^ mount_point)
           (fun () -> umount mount_point)
           () ;
+      let vbds = get_update_vbds ~__context ~vdi in
       Helpers.call_api_functions ~__context (fun rpc session_id ->
-          let dom0 = Helpers.get_domain_zero ~__context in
-          let vbds = Client.VDI.get_VBDs ~rpc ~session_id ~self:vdi in
           List.iter
             (fun self ->
-              if Client.VBD.get_VM ~rpc ~session_id ~self = dom0 then (
-                Client.VBD.unplug ~rpc ~session_id ~self ;
-                Client.VBD.destroy ~rpc ~session_id ~self
-              ))
+              Client.VBD.unplug ~rpc ~session_id ~self ;
+              Client.VBD.destroy ~rpc ~session_id ~self)
```

Comment:
I don't think this change has been explained in the commit message, what was the issue before?

Reply:
this is functionally equivalent to before (i believe) - i just factored out `get_update_vbds`